src/servers/src/grpc/greptime_handler.rs

```diff
@@ -98,12 +91,28 @@ impl GreptimeRequestHandler {
                     }
                     e
                 })
-        });
-
-        handle.await.context(JoinTaskSnafu).map_err(|e| {
-            timer.record(e.status_code());
-            e
-        })?
+        };
+
+        match &self.runtime {
+            Some(runtime) => {
+                // Executes requests in another runtime to
+                // 1. prevent the execution from being cancelled unexpected by Tonic runtime;
+                //   - Refer to our blog for the rational behind it:
+                //     https://www.greptime.com/blogs/2023-01-12-hidden-control-flow.html
+                //   - Obtaining a `JoinHandle` to get the panic message (if there's any).
+                //     From its docs, `JoinHandle` is cancel safe. The task keeps running even it's handle been dropped.
+                // 2. avoid the handler blocks the gRPC runtime incidentally.
```

Comment:
cancellation handler is still an "external cancel", and may violate (1) which brings unexpected (from the task's perspective) cancel